cirq-core/cirq/qis/clifford_tableau.py

```diff
@@ -176,6 +178,43 @@ def _str_full_(self) -> str:
 
         return string
 
+    def merged_with(self, second: 'CliffordTableau') -> 'CliffordTableau':
+        """Returns a merged CliffordTableau of self tableau and second tableau.
+
+        The meaning of merge two clifford tableau is the corresponding unitary operation
+        of merged clifford tableau is equal to (up to global phase) the composed unitary
+        operation of self tableau and the one of second tableau.
```

Comment:
This description is a little clunky, and I think it's essential to be very clear about what this does. "Merge" is a very ambiguous term. In fact: maybe this should be called `compose_with`?

"A merged [/composed] tableau is equal to (up to global phase) the composed unitary operation of the two tableaux"

Also say which one happens first, since I don't really know. Instead of "self", maybe say "this tableau" since it will be called `tableau.merged_with(tab2)`

Reply:
Updated the comments. Also, use the `then` as the function name as discussed in cyqn meeting.